README.Rmd

```diff
@@ -49,136 +48,85 @@ pak::pak("tidyverse/glue")
-This lets you indent the strings naturally in code.
 ```{r}
-glue("
-    A formatted string
-    Can have multiple lines
-      with additional indention preserved
-    ")
+x <- "the local environment"
+glue(
+  "`glue()` can access values from {x} or from {y}. {z}",
+  y = "named arguments",
+  z = "Woo!"
+)
 ```
 
-##### An additional newline can be used if you want a leading or trailing newline.
-```{r}
-glue("
-
-  leading or trailing newlines can be added explicitly
+If the relevant data lives in a data frame (or list or environment), use `glue_data()` instead:
 
-  ")
-```
-
-##### `\\` at the end of a line continues it without a new line.
 ```{r}
-glue("
-    A formatted string \\
-    can also be on a \\
-    single line
-    ")
+mini_mtcars <- head(cbind(model = rownames(mtcars), mtcars))
+glue_data(mini_mtcars, "{model} has {hp} hp.")
 ```
 
-##### A literal brace is inserted by using doubled braces.
-```{r}
-name <- "Fred"
-glue("My name is {name}, not {{name}}.")
-```
+`glue_data()` is very natural to use with the pipe:
 
-##### Alternative delimiters can be specified with `.open` and `.close`.
 ```{r}
-one <- "1"
-glue("The value of $e^{2\\pi i}$ is $<<one>>$.", .open = "<<", .close = ">>")
+mini_mtcars |>
+  glue_data("{model} gets {mpg} miles per gallon.")
 ```
 
-##### All valid R code works in expressions, including braces and escaping.
-Backslashes do need to be doubled just like in all R strings.
-```{r}
-  `foo}\`` <- "foo"
-glue("{
-      {
-        '}\\'' # { and } in comments, single quotes
-        \"}\\\"\" # or double quotes are ignored
-        `foo}\\`` # as are { in backticks
-      }
-  }")
-```
+These `glue_data()` examples also demonstrate that `glue()` is vectorized over the data.
 
-##### `glue_sql()` makes constructing SQL statements safe and easy
-Use backticks to quote identifiers, normal strings and numbers are quoted
-appropriately for your backend.
+### What you see is awfully close to what you get
 
-```{r}
-con <- DBI::dbConnect(RSQLite::SQLite(), ":memory:")
-colnames(iris) <- gsub("[.]", "_", tolower(colnames(iris)))
-DBI::dbWriteTable(con, "iris", iris)
-
-var <- "sepal_width"
-tbl <- "iris"
-num <- 2
-val <- "setosa"
-glue_sql("
-  SELECT {`var`}
-  FROM {`tbl`}
-  WHERE {`tbl`}.sepal_length > {num}
-    AND {`tbl`}.species = {val}
-  ", .con = con)
-```
+`glue()` lets you write code that makes it easy to predict what the final string will look like.
+There is considerably less syntactical noise and mystery, compared to `paste()` and `sprintf()`, for example.
 
-# Other implementations
+Empty first and last lines are automatically trimmed, as is leading whitespace that is common across all lines.
+You don't have to choose between indenting your code properly and getting the output you actually want.
 
-Some other implementations of string interpolation in R (although not using identical syntax).
+```{r}
+if (TRUE) {
```

Comment:
Why the `if (TRUE)`?

Reply:
Just to get some extra indentation and make the point about common leading whitespace being removed. In the vignette, I modified this example to put the `glue()` call inside a function body. I can do the same here to be less weird.